ds/reactive/set_test.go

```diff
@@ -75,3 +77,283 @@ func TestSubtract(t *testing.T) {
+		updatedElements = appliedMutations.AddedElements()
+	})
+	defer unsubscribe()
+
+	// Define a mutation factory
+	mutationFactory := func(set ds.ReadableSet[int]) ds.SetMutations[int] {
+		mutations := ds.NewSetMutations[int]()
+		if set.Has(2) {
+			mutations.WithDeletedElements(newSet(2)) // Delete element 2 if it exists
+		}
+		mutations.WithAddedElements(newSet(4)) // Add element 4
+		return mutations
+	}
+
+	// Apply the mutation
+	s.Compute(mutationFactory)
+
+	// Check if the callback was called and the mutations were applied correctly
+	require.True(t, callbackCalled, "OnUpdate callback should be called")
+	require.True(t, updatedElements.Has(4), "Element 4 should be added in the callback")
+}
+
+// TestSet_Replace_WithCallback tests the Replace method with a registered callback
+func TestSet_Replace_WithCallback(t *testing.T) {
+	s := newSet[int](1, 2, 3)
+	callbackCalled := false
+	removedElementsInCallback := ds.NewSet[int]()
+
+	// Register a callback
+	unsubscribe := s.OnUpdate(func(appliedMutations ds.SetMutations[int]) {
+		callbackCalled = true
+		removedElementsInCallback = appliedMutations.DeletedElements()
+	})
+	defer unsubscribe()
+
+	newElements := ds.NewSet[int](4, 5)
+
+	// Replace elements in the set
+	s.Replace(newElements.ReadOnly())
+
+	// Check if the callback was called and the elements are replaced correctly
+	require.True(t, callbackCalled, "OnUpdate callback should be called")
+	require.True(t, removedElementsInCallback.Has(1), "Element 1 should be removed in the callback")
+	require.True(t, removedElementsInCallback.Has(2), "Element 2 should be removed in the callback")
+	require.True(t, removedElementsInCallback.Has(3), "Element 3 should be removed in the callback")
+}
+
+// TestDerivedSet_inheritMutations_WithCallback tests the inheritMutations method with a registered callback
+func TestDerivedSet_inheritMutations_WithCallback(t *testing.T) {
+	dSet := newDerivedSet[int]()
+	callbackCalled := false
+	updatedElements := ds.NewSet[int]()
+
+	// Register a callback
+	unsubscribe := dSet.OnUpdate(func(appliedMutations ds.SetMutations[int]) {
+		callbackCalled = true
+		updatedElements = appliedMutations.AddedElements()
+	})
+	defer unsubscribe()
+
+	mutations := ds.NewSetMutations[int]().WithAddedElements(newSet(1, 2))
+
+	// Apply mutations
+	dSet.inheritMutations(mutations)
+
+	// Check if the callback was called and the mutations were applied correctly
+	require.True(t, callbackCalled, "OnUpdate callback should be called")
+	require.True(t, updatedElements.Has(1), "Element 1 should be added in the callback")
+	require.True(t, updatedElements.Has(2), "Element 2 should be added in the callback")
+}
+
+func TestSet_Encoding(t *testing.T) {
+	serix.DefaultAPI.RegisterTypeSettings("", serix.TypeSettings{}.WithLengthPrefixType(serix.LengthPrefixTypeAsByte))
+
+	testSet := newSet[string]()
+	for i := 0; i < 3; i++ {
+		testSet.Add(fmt.Sprintf("item%d", i))
+	}
+	bytes, err := testSet.Encode()
```

Comment:
mhh should the interface of the reactive `Set` maybe get an API passed to Encode and Decode?

Reply:
As the code is currently structured, this Encode is actually implemented by the OrderedMap!
We use the OrderedMap in way too many places to change that API and all the APIs that depend on it.